clang: Warning - destructor called on 'godot::CallableCustomMethodPointerBase'

### Godot version

current tip (4.2)

### godot-cpp version

current tip (4.2) 64eac01d04cd6ced10ad72dece92fdc971161d89

### System information

Debian clang version 16.0.6 / 17.0.3

### Issue description

+ Breaks a build with `env.Append(CCFLAGS='-Werror')`. 

```gdb
...
clang++ -o godot-cpp/src/variant/projection.linux.template_debug.dev.x86_64.o -c -std=c++17
-fPIC -Wwrite-strings -m64 -march=x86-64 -gdwarf-4 -g3 -Og -DHOT_RELOAD_ENABLED -DLINUX_ENABLED
-DUNIX_ENABLED -DDEBUG_ENABLED -DDEBUG_METHODS_ENABLED -DDEV_ENABLED
-Igodot-cpp/gdextension -Igodot-cpp/include
-Igodot-cpp/gen/include -Igodot-cpp/src godot-cpp/src/variant/projection.cpp

In file included from godot-cpp/src/variant/callable_method_pointer.cpp:31:
In file included from godot-cpp/include/godot_cpp/variant/callable_method_pointer.hpp:34:
In file included from godot-cpp/include/godot_cpp/core/binder_common.hpp:36:
In file included from godot-cpp/include/godot_cpp/core/method_ptrcall.hpp:36:
In file included from godot-cpp/include/godot_cpp/core/object.hpp:36:
In file included from godot-cpp/include/godot_cpp/core/property_info.hpp:38:
In file included from godot-cpp/include/godot_cpp/variant/variant.hpp:36:
In file included from godot-cpp/gen/include/godot_cpp/variant/builtin_types.hpp:36:
In file included from godot-cpp/gen/include/godot_cpp/variant/string.hpp:38:
In file included from godot-cpp/include/godot_cpp/variant/char_string.hpp:34:
In file included from godot-cpp/include/godot_cpp/templates/cowdata.hpp:37:
godot-cpp/include/godot_cpp/core/memory.hpp:102:3: warning: destructor called on
'godot::CallableCustomMethodPointerBase' that is abstract but has non-virtual destructor 
[-Wdelete-abstract-non-virtual-dtor]
                p_class->~T();
                ^
godot-cpp/src/variant/callable_method_pointer.cpp:44:2: note:
in instantiation of function template specialization
'godot::memdelete<godot::CallableCustomMethodPointerBase>' 
requested here
        memdelete(callable_method_pointer);
        ^
godot-cpp/include/godot_cpp/core/memory.hpp:102:13: note: qualify call to silence this warning
                p_class->~T();
                          ^
                          godot::CallableCustomMethodPointerBase::
clang++ -o godot-cpp/src/core/object.linux.template_debug.dev.x86_64.o -c -std=c++17 -fPIC -Wwrite-strings
-m64 -march=x86-64 -gdwarf-4 -g3 -Og -DHOT_RELOAD_ENABLED -DLINUX_ENABLED -DUNIX_ENABLED
-DDEBUG_ENABLED -DDEBUG_METHODS_ENABLED -DDEV_ENABLED -Igodot-cpp/gdextension
-Igodot-cpp/include -Igodot-cpp/gen/include -Igodot-cpp/src godot-cpp/src/core/object.cpp
 warning generated.
...
```

### Steps to reproduce

+ `scons use_llvm=yes dev_build=yes optimize=debug`



### Minimal reproduction project

N/A

--- a/include/godot_cpp/variant/callable_method_pointer.hpp
+++ b/include/godot_cpp/variant/callable_method_pointer.hpp
@@ -40,6 +40,7 @@ class CallableCustomMethodPointerBase {
 public:
 	virtual Object *get_object() const = 0;
 	virtual void call(const Variant **p_arguments, int p_argcount, Variant &r_return_value, GDExtensionCallError &r_call_error) const = 0;
+	virtual ~CallableCustomMethodPointerBase() {}
 };
 
 namespace internal {

